Needed for HEAD and glibc-2_7-branch.

Can't be installed as-is.

Discussed on <https://savannah.gnu.org/task/?7096>.

From Debian glibc repository.  Patch by Samuel Thibault.


r2425 of debian/patches/hurd-i386/local-no-strerror_l.diff.

Date: 2007-07-21 00:27:17 +0000 (Sat, 21 Jul 2007)
New Revision: 2425

  * hurd-i386/local-no-strerror_l.diff: new patch to disable non-implemented
    strerror_l().

Needs porting. Since this is new with 2.6, just disable for now, and please
someone take the time to port it.


---
 string/strerror_l.c |    3 +++
 string/string.h     |    3 ++-
 2 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/string/strerror_l.c b/string/strerror_l.c
index 32a004e..65dd5d5 100644
--- a/string/strerror_l.c
+++ b/string/strerror_l.c
@@ -23,6 +23,8 @@
 #include <string.h>
 #include <sys/param.h>
 
+#if 0
+Needs porting
 
 static __thread char *last_value;
 
@@ -70,3 +72,4 @@ text_set_element (__libc_thread_subfreeres, strerror_thread_freeres);
 text_set_element (__libc_subfreeres, strerror_thread_freeres);
 # endif
 #endif
+#endif
diff --git a/string/string.h b/string/string.h
index 90c92d4..301cca9 100644
--- a/string/string.h
+++ b/string/string.h
@@ -285,7 +285,8 @@ extern char *strerror_r (int __errnum, char *__buf, size_t __buflen)
 
 #ifdef __USE_GNU
 /* Translate error number to string according to the locale L.  */
-extern char *strerror_l (int __errnum, __locale_t __l) __THROW;
+/* Not yet ported on the Hurd.  */
+/* extern char *strerror_l (int __errnum, __locale_t __l) __THROW; */
 #endif
 
 
